python/plugins/processing/algs/qgis/TilesXYZ.py

```diff
@@ -182,8 +199,13 @@ def generate(self, writer, parameters, context, feedback):
         settings.setDestinationCrs(dest_crs)
         settings.setLayers(self.layers)
         settings.setOutputDpi(dpi)
-        if self.tile_format == 'PNG':
-            settings.setBackgroundColor(QColor(Qt.transparent))
+        canvas_red = QgsProject.instance().readNumEntry('Gui', '/CanvasColorRedPart', 255)[0]
```

Comment:
This needs to be done in the algorithm's "prepare" step so that it runs in the main thread. It also shouldn't use QgsProject.instance(), but rather take the project from the processing context given to prepareAlgorithm.